test/unit/math/src/math.cpp

```diff
@@ -42,19 +66,25 @@ struct TestKernel
     }
 };
 
-
 //#############################################################################
-// The TestTemplate runs the main code and the tests (Buffer,Functor,...).
-//! @tparam TAcc One of the possible accelerator types, that need to be tested.
-//! @tparam TData By now either double or float.
-template <
+template<
     typename TAcc,
-    typename TData>
+    typename TFunctor>
 struct TestTemplate
 {
-    template < typename TFunctor >
-    auto operator() ( unsigned long seed ) -> void
+    template<
+        typename TData>
+    auto operator()() -> void
     {
+        std::random_device rd{};
+        auto const seed = rd();
```

Comment:
We now randomly initialize the seed to get more coverage over time. I copied those two lines from the axpy test.